docs/guides/upgrade-apps-without-downtime.md

```diff
@@ -0,0 +1,182 @@
+# Upgrading Applications without Downtime
+
+This guide shows how the specific features of NGINX Kubernetes Gateway can help upgrade your applications without
+downtime.
+
+> We assume you are familiar with the upgrade methods mentioned in this guide. As a consequence, we will not cover
+> them in depth, but instead, focus on how NGINX Kubernetes Gateway supports them.
+<!--- This comment silences the linter. Otherwise, it will complain about the empty line, which is intentional here-->
+> By application downtime we mean the clients cannot get responses from your application. Instead,
+> they get responses with an error status code like 502 from NGINX.
+
+This guide covers:
```

Comment:
What do you think about adding every case considered in this guide to our test automation, so that it checks that an upgrade happens without downtime?

Reply:
Since we don't have test automation yet, do we have a list (or need a list) of potential tests?

Reply:
there is https://github.com/nginxinc/nginx-kubernetes-gateway/discussions/826 and also https://github.com/nginxinc/nginx-kubernetes-gateway/discussions/752

Reply:
added in https://github.com/nginxinc/nginx-kubernetes-gateway/discussions/826#discussioncomment-6705198